### ISSUE TITLE
[2.9 ACRA] App state crash - DeckPicker.showImportDialog after onSaveInstanceState

### ISSUE THREAD

**Issue by mikehardy** [ankidroid/Anki-Android#5504]
This one looks like a good mystery to untangle - happens to quite a few people

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/29951873-4d7b-4ee5-b2e1-1a96624b7b6b

```
java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
at androidx.fragment.app.FragmentManagerImpl.checkStateLoss(FragmentManagerImpl.java:2)
at androidx.fragment.app.FragmentManagerImpl.enqueueAction(FragmentManagerImpl.java:1)
at androidx.fragment.app.BackStackRecord.commitInternal(BackStackRecord.java:12)
at androidx.fragment.app.BackStackRecord.commit(BackStackRecord.java:1)
at androidx.fragment.app.DialogFragment.show(DialogFragment.java:10)
at com.ichi2.anki.AnkiActivity.showDialogFragment(AnkiActivity.java:5)
at com.ichi2.anki.DeckPicker.showImportDialog(DeckPicker.java:13)
at com.ichi2.anki.dialogs.DialogHandler.handleMessage(DialogHandler.java:6)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:214)
at android.app.ActivityThread.main(ActivityThread.java:7356)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

**Comment by mikehardy** [comment]
This is easily the plurality of our crashes

**Comment by mikehardy** [comment]
So, because we pop fragments in dialogs async, it's always possible the user can hit the home button or anything between the time we start working, and the time the dialog can pop. Which means we're past 'RESUME' state in the lifecycle, and shouldn't do any dialog fragment displays.

What we should do is check where we are in the lifecycle.
1. If we are in RESUME state, we should show the dialog
1. If we are not, we should add the dialog to a list of things to show
1. Then we should set a lifecycle listener on the RESUME state that looks at the list, and shows dialogs if they exist

Here's an example, Kotlin vs Java but still applies https://stackoverflow.com/a/46568517/9910298

**Comment by mikehardy** [comment]
This was fixed in #5743
